### PR TITLE
Supporting delete in db client

### PR DESCRIPTION
**Pull request by ahmedhamodi**
If unable to delete, error message propagated:
image

If successful (this is just hardcoded in tornado client side, can change):
image

If deleting a non-existent id (example, asset id 5 after it was already deleted):
image

### CODE REVIEW

**Review by drbalar — APPROVED** [review]
LG

**Comment by jaydhulia** [comment]
> Todo:
> 
> * explore exception error and possibly propagating that up instead of a boolean

Are you going to be adding the TODO to this PR and then review, or is that a TODO for a later date? @ahmedhamodi

**Comment by ahmedhamodi** [comment]
I was going to do the todo in this PR probably tomorrow (today since it's after midnight).

File: app/library/postgres.py [hunk 1]
```diff
@@ -72,9 +73,13 @@ async def get_asset_from_postgres(asset_id: str):
 
 async def delete_asset_from_postgres(asset_id: str):
     # TODO: delete from postgres
+    try:
+        delete_asset(asset_id)
+    except Exception as e:
+        raise e
```
Comment (jaydhulia) [inline app/library/postgres.py:79]: Let's do `delete_asset(asset_id)` just as is here, so we don't slow down the response by catching the exception and then propagating it.
Alternatively, if we want to make this more DCC friendly, we should catch it and turn it human readable:
```
try:
        delete_asset(asset_id)
    except psycopg2.errors.ForeignKeyViolation as e:
        raise AssertionError("One or more scenes are referencing this asset, please delete the scene before deleting this asset")
```

File: app/library/postgres.py [hunk 2]
```diff
@@ -72,9 +73,13 @@ async def get_asset_from_postgres(asset_id: str):
...
 
-    sample_response = {"message": "deleted successfully"}
-    return sample_response
+    response = {"message": "deleted successfully"}
```
Comment (jaydhulia) [inline app/library/postgres.py:81]: If an asset doesn't exist in the table, this method still returns `{"message": "deleted successfully"}`. Should we do a check to see if the asset exists first? Or is there a better way of distinguishing between non-valid asset IDs and valid asset IDs that are deleted?

File: app/models/db_client.py [hunk 3]
```diff
@@ -46,6 +46,18 @@ def get_asset(id):
     return asset
 
 
+# TODO: move session creating and closing out of these individual calls
+def delete_asset(id):
+    try:
+        session = Session()
+        session.query(Asset).filter(Asset.id == id).delete(synchronize_session="fetch")
```
Comment (jaydhulia) [inline app/models/db_client.py:53]: Out of curiousity, what does `synchronize_session="fetch"` do?

File: app/library/postgres.py (hunk 2 above)
Comment (ahmedhamodi) [reply app/library/postgres.py:81]: We shouldn't really allow the deletion of non-valid asset IDs in the frontend UI. But either way, I agree a different error message should be returned. I'll look into that

File: app/models/db_client.py (hunk 3 above)
Comment (ahmedhamodi) [reply app/models/db_client.py:53]: As per discussion offline:
image

Will investigate if delete needs to be synchronized or not.

File: app/library/postgres.py [hunk 4]
```diff
@@ -74,10 +75,13 @@ async def get_asset_from_postgres(asset_id: str):
...
-    # TODO: delete from postgres
+    did_delete = delete_asset(asset_id)
 
-    sample_response = {"message": "deleted successfully"}
-    return sample_response
+    if did_delete:
+        response = {"message": "deleted successfully"}
+    else:
+        response = {"message": "asset id " + asset_id + " not found"}
+    return response
```
Comment (ahmedhamodi) [inline app/library/postgres.py:84]: This is just an example of how I envision the deletion being used, but this is for @jaydhulia and @vivianliu0 to implement and change as needed.

**Comment by ahmedhamodi** [comment]
Note: leaving the cleaning of error message to your implementation/decision on Tornado side @jaydhulia @vivianliu0

**Review by jaydhulia — APPROVED** [review]

**Pull request merged**